pkg/apis/apps/v1beta2/defaults.go

```diff
@@ -68,7 +68,8 @@ func SetDefaults_StatefulSet(obj *appsv1beta2.StatefulSet) {
 	}
 
 	if obj.Spec.UpdateStrategy.Type == "" {
-		obj.Spec.UpdateStrategy.Type = appsv1beta2.OnDeleteStatefulSetStrategyType
+		obj.Spec.UpdateStrategy.Type = appsv1beta2.RollingUpdateStatefulSetStrategyType
+		obj.Spec.UpdateStrategy.RollingUpdate = &appsv1beta2.RollingUpdateStatefulSetStrategy{}
```

Comment:
Can you add a comment that this will default below to specific values? I would group the update strategy defaulting code.